src/main/java/graphql/util/NodeMultiZipper.java

```diff
@@ -24,11 +24,27 @@
     private final NodeAdapter<T> nodeAdapter;
 
     public NodeMultiZipper(T commonRoot, List<NodeZipper<T>> zippers, NodeAdapter<T> nodeAdapter) {
+        this.commonRoot = assertNotNull(commonRoot);
+        this.zippers = Collections.unmodifiableList(new ArrayList<>(zippers));
+        this.nodeAdapter = nodeAdapter;
+    }
+
+    /*
+     * constructor without defensive copy of the zippers
+     */
+    private NodeMultiZipper(T commonRoot, List<NodeZipper<T>> zippers, NodeAdapter<T> nodeAdapter, Object dummy) {
```

Comment:
This is what private methods are for.  You can TRUST yourself by the fact that it IS private!

(/)